tableauserverclient/server/endpoint/datasources_endpoint.py

```diff
@@ -284,7 +319,14 @@ def publish(
             return new_datasource
 
     @api(version="3.13")
-    def update_hyper_data(self, datasource_or_connection_item, *, request_id, actions, payload = None):
+    def update_hyper_data(
+        self,
+        datasource_or_connection_item: Union[DatasourceItem, ConnectionItem, str],
+        *,
+        request_id: str,
+        actions: Sequence[Mapping[str, Union[str, Mapping[str, str]]]],
```

Comment:
Can you make this into a typevar called `HyperActions` or something? My brain exploded reading that. (The new union syntax in 3.10 will be nice when we can use it)

It may also be worth turning that into a formal object at some point in the future cc @vogelsgesang